src/logic/contracts/methodIds.ts

```diff
@@ -59,39 +60,28 @@ const METHOD_TO_ID = {
 export type SafeMethods = typeof SAFE_METHODS_NAMES[keyof typeof SAFE_METHODS_NAMES]
 type TokenMethods = 'transfer' | 'transferFrom' | 'safeTransferFrom'
 
-type DecodedValues = Array<{
-  name: string
-  type?: string
-  value: string
-}>
-
 type SafeDecodedParams = {
-  [key in SafeMethods]?: DecodedValues
+  [key in SafeMethods]?: Record<string, string>
```

Comment:
Why was this changed from
```
type DecodedValues = Array<{	
  name: string	
  type?: string	
  value: string	
}>	
``` 
to more loosely typed 
```
Record<string, string>
```
?

Reply:
because that was wrong.

Reply:
but to me it seems that all the decodings still follow the same structure?
```
{ name: 'newOwner', type: 'address', value: decodedParameters[2] },
```

Reply:
yes... there's a big confusion around names. The returned type is `DataDecoded` which has that structure you're pointing here. And corresponds to the existing `dataDecoded` value returned by the backend, and is also the same structure we build in the app (this is kinda legacy code, but is used by the _pending_ txs) by calling `decodeMethods`.

```typescript
export interface DataDecoded {
  method: DataDecodedMethod
  parameters: Parameter[]
}
```

On the other hand, is `decodedParams`, that uses the `DecodedParam` type created in this file (messy and confusing all this) and is the structure already existing in the Safe, that is used by that not so good loved `getTxData` function, used to classify txs.

---

So, bottom line, I'll try to move the things around a bit, so it's a bit more explicit. But typing is fine.